app/assets/javascripts/settings.js.erb

```diff
@@ -27,8 +27,42 @@ See doc/COPYRIGHT.rdoc for more details.
 
 ++#%>
 
-jQuery(document).ready(function($) {
-  $('#settings_session_ttl_enabled').on('change', function(){
-    $('#settings_session_ttl_container').toggle($(this).is(':checked'));
-  }).trigger('change');
-});
+(function($) {
+  $(function() {
+
+    $('#settings_session_ttl_enabled').on('change', function(){
+      $('#settings_session_ttl_container').toggle($(this).is(':checked'));
+    }).trigger('change');
+
+
+    /** Sync SCM vendor select when enabled SCMs are changed */
+    $('[name="settings[enabled_scm][]"]').change(function () {
+      var checked = this.checked,
+        value = this.value,
+        select = $('#settings_repositories_automatic_managed_vendor'),
+        option = select.find('option[value="' + value + '"]'),
+        makeOption = function() {
+          return ['<option value="', value, '">', value, '</option>'].join('');
```

Comment:
I somehow remember it to be faster, however that information seems to be wrong (or outdated). I've changed it along with the r4 rebase.